internal/appsec/remoteconfig.go

```diff
@@ -0,0 +1,150 @@
+		log.Debug("appsec: Remote config: %d configs received for ASM_FEATURES. Expected one at most, returning early", l)
+		return statuses
+	}
+	for path, raw := range u {
+		var data rc.ASMFeaturesData
+		status := rc.ApplyStatus{State: rc.ApplyStateAcknowledged}
+		var err error = nil
+		log.Debug("appsec: Remote config: processing %s", path)
+
+		// A nil config means ASM was disabled, and we stopped receiving the config file
+		// Don't ack the config in this case and return early
+		if raw == nil {
+			log.Debug("appsec: Remote config: Stopping AppSec")
+			a.stop()
+			return statuses
+		}
+		if err = json.Unmarshal(raw, &data); err != nil {
+			log.Debug("appsec: Remote config: error while unmarshalling %s. Configuration won't be applied.", path)
+		} else if data.ASM.Enabled && !a.started {
+			log.Debug("appsec: Remote config: Starting AppSec")
+			if err = a.start(); err != nil {
+				log.Debug("appsec: Remote config: error while processing %s. Configuration won't be applied: %v", path, err)
+			}
+		} else if !data.ASM.Enabled && a.started {
+			log.Debug("appsec: Remote config: Stopping AppSec")
+			a.stop()
+		}
+		if err != nil {
+			status.State = rc.ApplyStateError
+			status.Error = err.Error()
+			log.Debug("appsec: Remote config: %s", status.Error)
+		}
+		statuses[path] = status
+	}
+
+	return statuses
+}
+
+func (a *appsec) startRC() {
+	if a.rc != nil {
+		a.rc.Start()
+	}
+}
+
+func (a *appsec) stopRC() {
+	if a.rc != nil {
+		a.rc.Stop()
+	}
+}
+
+func (a *appsec) registerRCProduct(product string) error {
+	if a.rc == nil {
+		return fmt.Errorf("no valid remote configuration client")
+	}
+	// Don't do anything if the product is already registered
+	for _, p := range a.rc.Products {
+		if p == product {
+			return nil
+		}
+	}
+	a.rc.Products = append(a.rc.Products, product)
+	return nil
+}
+func (a *appsec) registerRCCapability(c remoteconfig.Capability) error {
+	if a.rc == nil {
+		return fmt.Errorf("no valid remote configuration client")
+	}
+	// Don't do anything if the capability is already registered
+	for _, cap := range a.rc.Capabilities {
+		if cap == c {
+			return nil
+		}
+	}
+	a.rc.Capabilities = append(a.rc.Capabilities, c)
+	return nil
+}
+
+func (a *appsec) registerRCCallback(c remoteconfig.Callback, product string) error {
+	if a.rc != nil {
+		return fmt.Errorf("no valid remote configuration client")
+	}
+	a.rc.RegisterCallback(c, product)
+	return nil
+
+}
+
+func (a *appsec) enableRemoteActivation() error {
+	if a.rc == nil {
+		return fmt.Errorf("no valid remote configuration client")
+	}
+	// First verify that the WAF is in good health. We perform this check in order not to  falsely "allow" users to
+	// activate ASM through remote config if activation would fail when trying to register a WAF handle
+	// (ex: if the service runs on an unsupported platform).
+	if err := waf.Health(); err != nil {
+		log.Debug("appsec: WAF health check failed, remote activation will be disabled: %v", err)
```

Comment:
This log is fine to have in debug mode as it is part of the feature's behaviour so I don't consider it as an "unexpected error" that should be shown to the user.